Source/Urho3D/Scene/Serializable.cpp

```diff
@@ -678,6 +678,44 @@ bool Serializable::LoadFile(const ea::string& resourceName)
     return Load(realResourceName);
 }
 
+bool Serializable::SaveFile(const ea::string& destinationPath)
+{
+    return SaveFile(destinationPath,"RootNode");
+}
+bool Serializable::SaveFile(const ea::string& destinationPath, const ea::string& rootNode)
+{
+    //check if path is not empty string path
+    if (destinationPath.empty())
+    {
+        URHO3D_LOGERROR("Can not save to null path");
+        return false;
+    }
+
+    ResourceCache* cache = GetSubsystem<ResourceCache>();
+     
+    ea::string extension = GetExtension(destinationPath);
+
+    if (extension == ".xml")
+    {
+        XMLFile xmlFile = XMLFile(context_);
+        auto xmlElem = xmlFile.CreateRoot(rootNode);
+        if (SaveXML(xmlElem))
+        {
+            return  xmlFile.SaveFile(destinationPath);
+        }
+    }
+    if (extension == ".json")
```

Comment:
Should be `else if` due to obvious reasons. Also saving as binary is missing.

Reply:
omitted for now dues to unexpected results